pennylane/transforms/mitigate.py

```diff
@@ -318,6 +318,47 @@ def richardson_extrapolate(x, y):
     return poly_extrapolate(x, y, len(x) - 1)
 
 
+def exponential_extrapolate(x, y, asymptote=None, eps=1.0e-6):
+    r"""Extrapolate to the zero-noise limit using an exponential model (:math:`Ae^{Bx} + C`). This
+    is done by linearizing the data using a logarithm, whereupon a linear fit is performed. Once
+    the model parameter are found, they are transformed back to exponential parameters.
+
+    Args:
+        x (Array): Data in x
+        y (Array): Data in y = f(x)
+        asymptote (float): Infinite noise limit expected for your circuit of interest. Defaults to
+            0 in the case an asymptote is not supplied.
+        eps (float): Epsilon to regularize log(y - asymptote) when the argument is to close to
+            zero or negative.
+
+    Returns:
+        float: Extrapolated value at f(0).
+
+    .. seealso:: :func:`~.pennylane.transforms.richardson_extrapolate`, :func:`~.pennylane.transforms.mitigate_with_zne`
+
+    **Example:**
+
+    >>> x = np.linspace(1, 10, 5)
+    >>> y = np.exp(-x) + np.random.normal(scale=0.1, size=len(x))
+    >>> qml.transforms.exponential_extrapolate(x, y)
+    1.0955721717596099
```

Comment:
```suggestion
    **Example:**

    >>> np.random.seed(0)
    >>> x = np.linspace(1, 10, 5)
    >>> y = np.exp(-x) + np.random.normal(scale=0.1, size=len(x))
    >>> qml.transforms.exponential_extrapolate(x, y)
    0.23365009000522544
```

I was thinking we can add a seed here for reproducibility.